### ISSUE TITLE
CLI: deploy detect default branch of repo if not set

### ISSUE THREAD

**Issue by KiaraGrouwstra** [woodpecker-ci/woodpecker#1097]
### Clear and concise description of the problem

`woodpecker-cli deploy --help` now tells us:
```
   --branch value            branch filter (default: "master")
```

the git community tho seems to have been moving from the name `master` to `main`.

### Suggested solution

it would be nice if woodpecker allowed configuring the default branch (and update cli documentation) to reflect this.

### Alternative

default to `main` over `master`

### Additional context

related thread on gitea: #480

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by anbraten** [comment]
I would suggest that the command queries the repo endpoint first to get the default branch of a repo and use that in case the user hasn't passed --branch. Was that what you suggested as well?

**Comment by KiaraGrouwstra** [comment]
that sounds sane to me, yeah.
if that's already the present behavior then this may be mostly about how the cli documents the current situation.

**Comment by anbraten** [comment]
was solved by #3867
